subsys/net/lib/lwm2m/lwm2m_engine.h

```diff
@@ -30,63 +30,286 @@
 /* length of time in milliseconds to wait for buffer allocations */
 #define BUF_ALLOC_TIMEOUT K_SECONDS(1)
 
+/**
+ * @brief Used for debugging to print ip addresses.
+ *
+ * @param addr sockaddr for socket using ipv4 or ipv6.
+ * @return char*
+ */
 char *lwm2m_sprint_ip_addr(const struct sockaddr *addr);
+
+/**
+ * @brief Converts the token to a printable format.
+ *
+ * @param[in] token Token to be printed.
+ * @param tkl Lenghts of token.
```

Comment:
Please be consistent, either document whether the parameter is input or output for all parameters, or none.